### PR TITLE
bugfix:when dfget is cdn source,supernode will not work

### PR DESCRIPTION
**Pull request by hb5488**
### Ⅰ. Describe what this PR did
当dfget 端口启动失败时，会默认走cdnsource方式，但是supernode对cdnsource方式支持不够完善，主要是增强cdnsource方式
1，当为cdnsource方式默认不进行调度
2，当cdnsource下载完成，调用service/down逻辑

### Ⅱ. Does this pull request fix one issue?

NONE

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)

不需要

### Ⅳ. Describe how to verify it

通过dfget 启动加入cdn=source方式进行下载
### Ⅴ. Special notes for reviews

目前master分支没有合并前就不能够正常下载

### CODE REVIEW

**Comment by hb5488** [comment]
@jim3ma 这个pr还有啥问题么？

File: apis/types/task_create_request.go [hunk 1]
```diff
@@ -101,6 +101,8 @@ type TaskCreateRequest struct {
 	// --filter parameter of dfget. The usage of it is that different rawURL may generate the same taskID.
 	//
 	TaskURL string `json:"taskURL,omitempty"`
+	// peer Pattern p2p or cdn
+	PeerPattern int32
```
Comment (jim3ma) [inline apis/types/task_create_request.go:105]: omitempty

File: supernode/config/constants.go [hunk 2]
```diff
@@ -136,3 +136,9 @@ const (
 	// unit: MB/s
 	DefaultMaxBandwidth = 200 * rate.MB
 )
+
+// Pattern
+const (
+	P2pPattern = int32(0)
```
Comment (jim3ma) [inline supernode/config/constants.go:142]: make it to new type `Pattern` is better to understand

**Comment by jim3ma** [comment]
@hb5488 feature is ok, I think some type need to optimize from comment.

File: supernode/config/constants.go (hunk 2 above)
Comment (hb5488) [reply supernode/config/constants.go:142]: DefaultPattern？ or P2pPattern = int32(1) not int32(0)

Comment (jim3ma) [reply supernode/config/constants.go:142]: type Pattern int32
const (
	P2pPattern = Pattern(0)
)
func x(p Pattern) {
}

**Comment by hb5488** [comment]
> @hb5488 feature is ok, I think some type need to optimize from comment.

done

**Comment by jim3ma** [comment]
LGTM

**Pull request merged**